### PR TITLE
custom altitude for (twilight) finder

### PR DESCRIPTION
**Pull request by axd1967**
### Description

This PR allows the user to define a local horizon when searching for rise/settings of objects.

Fixes # (issue)

### Screenshots (if appropriate):

image

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] This change requires a documentation update

### How Has This Been Tested?

**Test Configuration**:

Operating System: Kubuntu 20.04
KDE Plasma Version: 5.18.5
KDE Frameworks Version: 5.68.0
Qt Version: 5.12.8
Kernel Version: 5.4.0-86-generic
OS Type: 64-bit
Processors: 8 × Intel® Core™ i7-10510U CPU @ 1.80GHz
Memory: 15.3 GiB of RAM

## Checklist:

- [x] My code follows the [code style](http://stellarium.org/doc/head/codingStyle.html) of this project.
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: src/core/modules/Planet.cpp [hunk 1]
```diff
@@ -4382,23 +4382,31 @@ Vec4d Planet::getRTSTime(const StelCore *core, const double altitude) const
...
 		//ho = -0.25*asin(6378.14/(eclipticPos.length()*AU)); // horizon parallax factor.
 		ho = - getAngularSize(core) * M_PI_180; // semidiameter;
 	else if (getEnglishName()=="Sun")
 		ho = - getAngularSize(core) * M_PI_180; // semidiameter; Canonical value 16', but this is accurate even from other planets...
 
+	// include atmospheric refraction
 	if (core->getSkyDrawer()->getFlagHasAtmosphere())
 	{
-		// canonical" refraction at horizon is -34'. Replace by pressure-dependent value here!
+		// canonical" refraction at horizon is -34'. Replace by pressure-dependent value here! TODO
```
Comment (gzotti) [inline src/core/modules/Planet.cpp:4397]: Not a TODO, but done right here.

**Comment by gzotti** [comment]
Thanks, but my next iteration on this topic will finally take the currently installed landscape into account. No more need for an extra manual setting which cannot adapt itself to steep horizons.

**Comment by axd1967** [comment]
Indeed! But this also works without landscapes.

**Comment by axd1967** [comment]
And it allows to search for other times, such as astronomical twilight events (by setting the altitude to the desired negative value).

**Comment by gzotti** [comment]
> And it allows to search for other times, such as astronomical twilight events (by setting the altitude to the desired negative value).

Can you imagine what the purpose of the existing GUI element is?

**Comment by axd1967** [comment]
> Can you imagine what the purpose of the existing GUI element is?

Sure, but for the Sun only.  My PR works for any selected object.

**Comment by alex-w** [comment]
This pull request looks meaningless for me, because rise and set times are always computed from the real or mathematical horizon. Of course horizon lines from the landscape are not taken into account at this step and we should use the mathematical horizon only (this is default behavior of the tool). Negative altitude is used only for the Sun to compute the twilights. I can imagine how it may be used for the Moon also, but definitely not for any other celestial objects. The definition of the negative altitude for the Sun and compute twilight is implemented in the other tool already.

**Comment by axd1967** [comment]
guys, you are just playing games.

guess who is paying the price? the user community that is deprived from a nice tool, just because you despise me.

for me the tool is working nice and helping me a lot. sorry for the rest of the community.

**Pull request closed without merging**